pydantic_ai_slim/pydantic_ai/settings.py

```diff
@@ -0,0 +1,30 @@
+from __future__ import annotations
+
+from httpx import Timeout
+from typing_extensions import TypedDict
+
+
+class ModelSettings(TypedDict, total=False):
+    """The runtime settings for a model's request / message creation."""
+
+    max_tokens: int
+    """The maximum number of tokens to generate before stopping."""
+
+    temperature: float
+    """Amount of randomness injected into the response.
+
+    Use `temperature` closer to `0.0` for analytical / multiple choice, and closer to a model's
+    maximum `temperature` for creative and generative tasks.
+
+    Note that even with `temperature` of `0.0`, the results will not be fully deterministic.
+    """
+
+    top_p: float
+    """An alternative to sampling with temperature, called nucleus sampling, where the model considers the results of the tokens with top_p probability mass.
+
+    So 0.1 means only the tokens comprising the top 10% probability mass are considered.
+
+    You should either alter `temperature` or `top_p`, but not both."""
+
+    timeout: float | Timeout
+    """Override the client-level default timeout for a request, in seconds"""
```

Comment:
It might make sense to move the logic used for merging away out of the code above and create a function for doing it here:
```suggestion
    """Override the client-level default timeout for a request, in seconds"""

def merge_settings(base: ModelSettings | None, overrides: ModelSettings | None) -> ModelSettings:
    # Note: we may want merge recursively if/when we add non-primitive values
    return {**(base or {}), **(overrides or {})}
```
This might make it easier to remember to consider (if not change) how things work if, e.g., we add model-specific settings in a dict, and want to support overriding individual values.

I'm okay if we don't do this, but I think it helps make the implementation choices more explicit either way.

Reply:
Yep, I like this idea.